packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/MarkerBuilder.java

```diff
@@ -7,23 +7,54 @@
 import com.google.android.gms.maps.model.BitmapDescriptor;
 import com.google.android.gms.maps.model.LatLng;
 import com.google.android.gms.maps.model.MarkerOptions;
+import com.google.maps.android.clustering.ClusterItem;
 
-class MarkerBuilder implements MarkerOptionsSink {
+class MarkerBuilder implements MarkerOptionsSink, ClusterItem {
   private final MarkerOptions markerOptions;
+  private String clusterManagerId;
+  private String markerId;
   private boolean consumeTapEvents;
 
-  MarkerBuilder() {
+  MarkerBuilder(String markerId, String clusterManagerId) {
     this.markerOptions = new MarkerOptions();
+    this.markerId = markerId;
+    this.clusterManagerId = clusterManagerId;
   }
 
   MarkerOptions build() {
     return markerOptions;
   }
 
+  /** Update existing markerOptions with builder values */
+  void update(MarkerOptions markerOptionsToUpdate) {
```

Comment:
So that I can understand can you explain how markerOptionsToUpdate updates information?

Reply:
Google maps markers are created with MarkerOptions object.
Items added to clusterManager are instances of MarkerBuilder that extends the ClusterItem class. The clustering process uses data from this MarkerBuilder instance to update the MarkerOptions, that it then uses to create or re-create the marker to the map.

It is important to store this data to MarkerBuilder object instance per each marker to allow to update marker data even when marker instance is not drawn on the map.
Here is clustering callback that uses this update method tuo pass the data from the MarkerBuilder to the markerOptions instance, the clustermanager uses to create the visible marker on the map.

```java
    @Override
    protected void onBeforeClusterItemRendered(
        @NonNull MarkerBuilder item, @NonNull MarkerOptions markerOptions) {
      // Builds new markerOptions for new marker created by the ClusterRenderer under
      // ClusterManager.
      item.update(markerOptions);
    }
```